packages/provider/src/wallet.ts

```diff
@@ -728,6 +732,20 @@ export const useWallet = () => {
   return walletInstance
 }
 
-export const resetWallet = () => {
-  walletInstance = undefined
+export const connectWallet = async (network?: string | number, options?: ConnectOptions & { walletConfig?: Partial<ProviderConfig> }) => {
+  if (!walletInstance) {
+    walletInstance = getWallet(network, options?.walletConfig)
+  }
+  if (network && options) {
+    options.networkId = network
+  }
+  const connectDetails = await walletInstance.connect(options)
+  return { wallet: walletInstance, connectDetails }
+}
+
+export const disconnectWallet = () => {
```

Comment:
this is never safe, we can't dispose of the wallet, otherwise getWallet will crash when we call it next, because we can't duplicate the transport

Reply:
this is the same as `wallet.disconnect()` which a dapp certainly may want to use. it's just a short-hand.. but we could remove it and force devs to just .. `sequence.getWallet().disconnect()` .. I think that is fine too, so yea will remove this method